### PR TITLE
fix: `publish_dataset` to check `required=True` on at least one field/question

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR fixes a bug with the `PUT /api/v1/datasets/{dataset_id}/publish` endpoint, as there was a missing check on the `required` flag for each field/question which was allowing to publish `FeedbackTask` datasets with no required fields and/or questions.

**Type of change**

- [X] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**

- [X] Update existing unit/integration tests
- [X] Add unit/integration tests with `required=True` and `required=False`

**Checklist**

- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: tests/unit/server/api/v1/test_datasets.py [hunk 1]
```diff
@@ -3183,19 +3183,45 @@ async def test_publish_dataset_without_fields(
...
+        assert (await db.execute(select(func.count(Record.id)))).scalar() == 0
+
+    async def test_publish_dataset_without_required_questions(
+        self, async_client: "AsyncClient", db: "AsyncSession", owner_auth_header: dict
+    ) -> None:
+        dataset = await DatasetFactory.create()
+        await TextFieldFactory.create(dataset=dataset, required=True)
+        await RatingQuestionFactory.create(dataset=dataset)
 
         response = await async_client.put(f"/api/v1/datasets/{dataset.id}/publish", headers=owner_auth_header)
```
Comment (gabrielmbmb) [inline tests/unit/server/api/v1/test_datasets.py]: We should just keep 1 of these unit tests, they are testing the same thing

**Review by gabrielmbmb — APPROVED** [review]
LGTM!

File: tests/unit/server/api/v1/test_datasets.py (hunk 1 above)
Comment (alvarobartt) [reply tests/unit/server/api/v1/test_datasets.py]: It was to ensure that the error is being raised on both no questions (former check) and no required questions (updated check)

Comment (alvarobartt) [reply tests/unit/server/api/v1/test_datasets.py]: Maybe we can just parametrize the `required` value for both

**Pull request merged**